### ISSUE TITLE
Non-collapsible `if` with `assert` is detected

### ISSUE THREAD

**Issue by MKowalski8** [software-mansion/cairo-lint#227]
### Problem

collapsible_if diagnoses this code as wrong, even though it is correct.
``` rust 
if x == y {
    assert!(z == Some(42));
}
```

### Scarb Version

```text
2.11.0
```

### Notes

The problem occurs inside this [if_chain](https://github.com/software-mansion/cairo-lint/blob/8bb853f790e6f0bc785aacbdf448d99f684821b8/crates/cairo-lint-core/src/lints/ifs/collapsible_if.rs#L100).

**Comment by wawel37** [comment]
Resolved by: https://github.com/software-mansion/cairo-lint/pull/260
